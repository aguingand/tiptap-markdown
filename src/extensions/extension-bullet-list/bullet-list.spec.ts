import { expect, test } from "vitest";
import { parse, serialize } from "../../../__tests__/utils";


test('parse markdown with "-"', () => {
    expect(parse('- example1\n\n- example2')).toMatchSnapshot();
});
test('parse markdown with "*"', () => {
    expect(parse('* example1\n\n* example2')).toMatchSnapshot();
});
test('parse markdown with "+"', () => {
    expect(parse('+ example1\n\n+ example2')).toMatchSnapshot();
});
test('parse html', () => {
    expect(parse('<ul><li>example1</li><li>example2</li></ul>')).toMatchSnapshot();
});
test('serialize', () => {
    expect(serialize('<ul><li>example1</li><li>example2</li></ul>'))
        .toEqual('- example1\n- example2\n');

    expect(serialize('<ul><li>example1</li><li>example2</li></ul>', { markdown: { bulletListMarker: '*' } }))
        .toEqual('* example1\n* example2\n');
});
