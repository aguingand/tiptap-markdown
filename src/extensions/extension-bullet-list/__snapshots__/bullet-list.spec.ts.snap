// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`html 1`] = `
[
  {
    "attrs": {
      "tight": true,
    },
    "content": [
      {
        "content": [
          {
            "content": [
              {
                "text": "example1",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "listItem",
      },
      {
        "content": [
          {
            "content": [
              {
                "text": "example2",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "listItem",
      },
    ],
    "type": "bulletList",
  },
]
`;

exports[`markdown with "*" 1`] = `
[
  {
    "attrs": {
      "tight": false,
    },
    "content": [
      {
        "content": [
          {
            "content": [
              {
                "text": "example1",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "listItem",
      },
      {
        "content": [
          {
            "content": [
              {
                "text": "example2",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "listItem",
      },
    ],
    "type": "bulletList",
  },
]
`;

exports[`markdown with "+" 1`] = `
[
  {
    "attrs": {
      "tight": false,
    },
    "content": [
      {
        "content": [
          {
            "content": [
              {
                "text": "example1",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "listItem",
      },
      {
        "content": [
          {
            "content": [
              {
                "text": "example2",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "listItem",
      },
    ],
    "type": "bulletList",
  },
]
`;

exports[`markdown with "-" 1`] = `
[
  {
    "attrs": {
      "tight": false,
    },
    "content": [
      {
        "content": [
          {
            "content": [
              {
                "text": "example1",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "listItem",
      },
      {
        "content": [
          {
            "content": [
              {
                "text": "example2",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "listItem",
      },
    ],
    "type": "bulletList",
  },
]
`;
